js/piwik.js

```diff
@@ -3869,47 +3869,134 @@ if (typeof Piwik !== 'object') {
+                        return 'right';
+                }
+            }
+
+            function getTargetElementFromEvent(event)
+            {
+                return event.target || event.srcElement;
+            }
+
             /*
              * Handle click event
              */
-            function clickHandler(evt) {
-                var button,
-                    target;
+            function clickHandler(enable) {
 
-                evt = evt || windowAlias.event;
-                button = evt.which || evt.button;
-                target = evt.target || evt.srcElement;
+                return function (event) {
 
-                // Using evt.type (added in IE4), we avoid defining separate handlers for mouseup and mousedown.
-                if (evt.type === 'click') {
-                    if (target) {
-                        processClick(target);
-                    }
-                } else if (evt.type === 'mousedown') {
-                    if ((button === 1 || button === 2) && target) {
-                        lastButton = button;
-                        lastTarget = target;
-                    } else {
+                    event = event || windowAlias.event;
+
+                    var button = getNameOfClickedButton(event);
+                    var target = getTargetElementFromEvent(event);
+
+                    if (event.type === 'click') {
+
+                        var ignoreClick = false;
+                        if (enable && button === 'middle') {
+                            // if enabled, we track middle clicks via mouseup
+                            // some browsers (eg chrome) trigger click and mousedown/up events when middle is clicked,
+                            // whereas some do not. This way we make "sure" to track them only once, either in click
+                            // (default) or in mouseup (if enable == true)
+                            ignoreClick = true;
+                        }
+
+                        if (target && !ignoreClick) {
+                            processClick(target);
+                        }
+                    } else if (event.type === 'mousedown') {
+                        if (button === 'middle' && target) {
+                            lastButton = button;
+                            lastTarget = target;
+                        } else {
+                            lastButton = lastTarget = null;
+                        }
+                    } else if (event.type === 'mouseup') {
+                        if (button === lastButton && target === lastTarget) {
+                            processClick(target);
+                        }
                         lastButton = lastTarget = null;
-                    }
-                } else if (evt.type === 'mouseup') {
-                    if (button === lastButton && target === lastTarget) {
+                    } else if (event.type === 'contextmenu') {
                         processClick(target);
```

Comment:
It's important to know that when the `contextmenu` event is triggered, a `click` event is not triggered. At least in all the browsers that I tested. This is important as we would otherwise track a click multiple times. There is a certain risk if a browser changes this behaviour or if a browser is buggy that we track multiple times.